client/components/payment-status-chip/test/index.js

```diff
@@ -7,70 +7,33 @@ import { render } from '@testing-library/react';
 /**
  * Internal dependencies
  */
-import { getChargeStatus } from '../../../utils/charge';
 import PaymentStatusChip from '../';
 
-jest.mock( '../../../utils/charge', () => ( { getChargeStatus: jest.fn() } ) );
-
 describe( 'PaymentStatusChip', () => {
-	test( 'renders a default light chip with no message if status does not match', () => {
-		getChargeStatus.mockReturnValue( 'teststatus' );
-		expect( renderPaymentStatus() ).toMatchSnapshot();
-	} );
-
-	test( 'renders a light chip with partially refunded message if there\'s a partial refund', () => {
-		getChargeStatus.mockReturnValue( 'refunded_partial' );
-		expect( renderPaymentStatus() ).toMatchSnapshot();
-	} );
-
-	test( 'renders a light chip with fully refunded message if there\'s a full refund', () => {
-		getChargeStatus.mockReturnValue( 'refunded_full' );
-		expect( renderPaymentStatus() ).toMatchSnapshot();
-	} );
-
-	test( 'renders a light chip with paid message if it is paid', () => {
-		getChargeStatus.mockReturnValue( 'paid' );
-		expect( renderPaymentStatus() ).toMatchSnapshot();
-	} );
-
-	test( 'renders a primary chip with authorized message if payment was not captured', () => {
-		getChargeStatus.mockReturnValue( 'authorized' );
-		const paymentStatusChip = renderPaymentStatus();
-		expect( paymentStatusChip ).toMatchSnapshot();
-	} );
-
-	test( 'renders an alert chip with failed message if payment status is failed', () => {
-		getChargeStatus.mockReturnValue( 'failed' );
-		expect( renderPaymentStatus() ).toMatchSnapshot();
+	test( 'renders default appearance if status is unrecognized', () => {
+		const { container } = render( <PaymentStatusChip status="mock_status" /> );
+		expect( container ).toMatchSnapshot();
+	} );
+
+	const statuses = [
+		'refunded_partial',
+		'refunded_full',
+		'paid',
+		'authorized',
+		'failed',
+		'blocked',
+		'disputed_needs_response',
+		'disputed_under_review',
+		'disputed_won',
+		'disputed_lost',
+		'disputed_warning_needs_response',
+		'disputed_warning_under_review',
+		'disputed_warning_closed',
+	];
+
+	test.each( statuses )( 'renders %s status', ( status ) => {
+		const { container } = render( <PaymentStatusChip status={ status } /> );
+		expect( container ).toMatchSnapshot();
```

Comment:
I like how the use of `test.each` to remove the redundancy of the previous tests ❤️